src/details.rs

```diff
@@ -84,7 +84,7 @@ impl<'a> DetailsItem<'a> {
             None => Decimal::ZERO,
         };
 
-        self.quantity * self.unit_price / base_quantity + reduction_and_surcharge_sum
+        (self.quantity * self.unit_price / base_quantity + reduction_and_surcharge_sum).round_dp(2)
```

Comment:
`round_dp` uses `Bankersrounding`: 6.5 -> 6

We should use `clone_with_scale` here. It uses `MidpointAwayFromZero` rounding.

Also this may affect line_item_total_gross_amount results due to previous rounding.

I think we should round right when creating the TaxItem elements.

Reply:
I just changed it to use the round_dp_with_strategy to round with MidpointAwayFromZero strategy. I wanted to use this method instead of clone_with_scale because it is directly implemented by rust_decimal and does the same.
Regarding `line_item_total_gross_amount` I think it is totally fine to use rounded line items as we should always calculate with the values you see on the invoice in the end.